dvc/commands/ls/__init__.py

```diff
@@ -9,18 +9,26 @@
 logger = logger.getChild(__name__)
 
 
-def _format_entry(entry, fmt):
+def _format_entry(entry, fmt, with_size=True, with_md5=False):
     from dvc.utils.humanize import naturalsize
 
-    size = entry.get("size")
-    if size is None:
-        size = ""
-    else:
-        size = naturalsize(size)
-    return size, fmt(entry)
+    if with_size:
+        size = entry.get("size")
+        if size is None:
+            size = ""
+        else:
+            size = naturalsize(size)
+    if with_md5:
+        md5 = entry.get("md5", "")
+    if with_size and not with_md5:
+        return size, fmt(entry)
+    if with_md5 and not with_size:
+        return md5, fmt(entry)
+    if with_size and with_md5:
+        return size, md5, fmt(entry)
```

Comment:
Consider simplifying this `if` condition.

```suggestion
    ret = []
    if with_size:
        size = entry.get("size")
        if size is None:
            size = ""
        else:
            size = naturalsize(size)
        ret.append(size)
    if with_md5:
        md5 = entry.get("md5", "")
        ret.append(md5)
    ret.append(fmt(entry))
    return ret
```